src/main/java/com/xjeffrose/xio/http/RequestBuffer.java

```diff
@@ -0,0 +1,92 @@
+package com.xjeffrose.xio.http;
+
+import com.xjeffrose.xio.client.ClientState;
+import com.xjeffrose.xio.client.DefaultChannelInitializer;
+import com.xjeffrose.xio.client.XioClient;
+import com.xjeffrose.xio.client.XioClientBootstrap;
+import com.xjeffrose.xio.server.Route;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.handler.codec.http2.Http2DataFrame;
+import io.netty.handler.codec.http2.Http2Headers;
+import io.netty.util.AttributeKey;
+import io.netty.util.ReferenceCountUtil;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import java.util.function.BiFunction;
+
+@Slf4j
+public class RequestBuffer extends ChannelDuplexHandler {
+
+  public static class WriteReady {
+    public static final WriteReady INSTANCE = new WriteReady();
+  }
+
+  public static class ObjectAndPromise {
+    public final Object object;
+    public final ChannelPromise promise;
+
+    ObjectAndPromise(Object object, ChannelPromise promise) {
+      this.object = object;
+      this.promise = promise;
+    }
+
+    public ChannelFuture apply(BiFunction<Object, ChannelPromise, ChannelFuture> write) {
+      return write.apply(object, promise);
+    }
+  }
+
+  List<ObjectAndPromise> writeBuffer = new ArrayList<>();
+
+  @Override
+  public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+    if (!(evt instanceof WriteReady)) {
```

Comment:
Why not `evt == WriteReady.INSTANCE`?

Also, why not just:
```java
public static final Object WRITE_READY = new Object();
```
?

Reply:
Following netty convention here:
https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionPrefaceAndSettingsFrameWrittenEvent.java

Although they seem to be conflicted on equals versus instanceof 
https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java#L233-L238

With this:
```
  public static class WriteReady {
    public static final WriteReady INSTANCE = new WriteReady();
  }
```
If you have a message logger in your pipeline it will log that an event was fired of type `WriteReady`. You lose that information with `Object WRITE_READY`. I should probably make the constructor private.

There's probably a more in depth conversation to be had about the speed of `==` vs `instanceof`. I have no preference here nor the desire to micro benchmark at this point in time. :)

Reply:
Fair enough argument, I'll change to eq.